internal/datacoord/meta.go

```diff
@@ -495,7 +496,7 @@ func CreateL0Operator(collectionID, partitionID, segmentID int64, channel string
 // Set status of segment
 // and record dropped time when change segment status to dropped
 func UpdateStatusOperator(segmentID int64, status commonpb.SegmentState) UpdateOperator {
-	return func(modPack *UpdateSegmentPack) bool {
+	return func(modPack *updateSegmentPack) bool {
```

Comment:
> Don't suggest using this operator pipeline, it might produce a dirty segment meta if the executing operator fails. we should clone a segment meta before running pipeline, apply all changes to the cloned segment meta, and update the segment cache after running the pipeline finish.

SegmentPack make sure segment was clone from meta, but operator exactly not a good abstarction, i will try to refine this other pr.